index.js

```diff
@@ -50,16 +48,16 @@ module.exports = function (path, second, third, fourth) {
  *
  * @returns [void]
  */
-var asciify_core = function(path, opts, onSuccess, onFailure) {
+var asciify_core = function(path, opts, callback) {
   // First open image to get initial properties
   Jimp.read(path, function(err, image) {
-    if (err) return onFailure('Error loading image: ' + err);
+    if (err) return callback('Error loading image: ' + err);
 
     // Setup options
     var options = {
       fit:     opts.fit     ? opts.fit               : 'original',
-      width:   opts.width   ? parseInt(opts.width)   : image.bitmap.width,
-      height:  opts.height  ? parseInt(opts.height)  : image.bitmap.height,
+      width:   opts.width   ? (opts.width.substr(-1) === '%' ? Math.floor((parseInt(opts.width.slice(0, -1)) / 100) * (windowSize.width * terminalCharWidth)) : parseInt(opts.width)) : image.bitmap.width,
+      height:  opts.height  ? (opts.height.substr(-1) === '%' ? Math.floor((parseInt(opts.height.slice(0, -1)) / 100) * windowSize.height) : parseInt(opts.height)) : image.bitmap.height,
```

Comment:
Can you move this parsing either into a function or into a variable outside this object declaration? Gets hard to read in a single line like this.

Reply:
Totally understandable! I'll get that done today.